x-pack/plugins/observability_solution/asset_manager/server/lib/entities/uninstall_entity_definition.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { SavedObjectsClientContract } from '@kbn/core-saved-objects-api-server';
+import { EntityDefinition } from '@kbn/entities-schema';
+import { Logger } from '@kbn/logging';
+import { deleteEntityDefinition } from './delete_entity_definition';
+import { deleteHistoryIngestPipeline, deleteLatestIngestPipeline } from './delete_ingest_pipeline';
+import {
+  stopAndDeleteHistoryTransform,
+  stopAndDeleteLatestTransform,
+} from './stop_and_delete_transform';
+
+export async function uninstallEntityDefinition({
+  definition,
+  esClient,
+  soClient,
+  logger,
+}: {
+  definition: EntityDefinition;
+  esClient: ElasticsearchClient;
+  soClient: SavedObjectsClientContract;
+  logger: Logger;
+}) {
+  await stopAndDeleteHistoryTransform(esClient, definition, logger);
+  await stopAndDeleteLatestTransform(esClient, definition, logger);
+  await deleteHistoryIngestPipeline(esClient, definition, logger);
+  await deleteLatestIngestPipeline(esClient, definition, logger);
+  await deleteEntityDefinition(soClient, definition, logger);
```

Comment:
I'd leave this to the client to retry failures in the uninstall process. The errors are send back to client and if it is persistent after retries it can be surfaced in the UI (or investigated in logs)

Reply:
Well, that would be a re-try for the thing that fails but it means we don't clean up other things in the meantime.

So imagine that the first two steps succeed, but step 3 fails, then we don't try to do step 4 and 5 until step 3 succeeds (in a re-try by the user at a later time).
That means those resources live on until the users resolves step 3 but they could already be deleted in the first try.